example/pubspec.yaml

```diff
@@ -1,44 +1,65 @@
 name: posthog_flutter_example
 description: Demonstrates how to use the posthog_flutter plugin.
-publish_to: "none"
-
-version: 0.0.1+1
+# The following line prevents the package from being accidentally published to
+# pub.dev using `flutter pub publish`. This is preferred for private packages.
+publish_to: "none" # Remove this line if you wish to publish to pub.dev
 
 environment:
-  sdk: '>=2.12.0 <4.0.0'
-  flutter: '>=1.10.0'
+  sdk: '>=2.18.0 <4.0.0'
+  flutter: '>=3.3.0'
 
+# Dependencies specify other packages that your package needs in order to work.
+# To automatically upgrade your package dependencies to the latest versions
+# consider running `flutter pub upgrade --major-versions`. Alternatively,
+# dependencies can be manually updated by changing the version numbers below to
+# the latest version available on pub.dev. To see which dependencies have newer
+# versions available, run `flutter pub outdated`.
 dependencies:
   flutter:
     sdk: flutter
+
   posthog_flutter:
+    # When depending on this package from a real application you should use:
+    #   posthog_flutter: ^x.y.z
+    # See https://dart.dev/tools/pub/dependencies#version-constraints
+    # The example app is bundled with the plugin so we use a path dependency on
+    # the parent directory to use the current plugin's version.
     path: ../
 
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
-  cupertino_icons: ^1.0.3
+  cupertino_icons: ^1.0.2
```

Comment:
does it matter this is a downgrade?

Reply:
Nope, `^` gets the latest version anyway, the tooling did it automatically.

Reply:
I reverted just because it was done by the tooling anyway